### PR TITLE
Cleanup plant solver and manager

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
 A relatively quick refactor of the plant manager and plant solver, along with utility functions. Ultimately this will result in a much nicer plant management structure with functions properly contained as class methods.

### CODE REVIEW

**Comment by Myoldmopar** [comment]
The files changed just exploded because I moved DataPlant into Plant/DataPlant where it belongs.  I could continue refactoring the plant code for a long time, but honestly, if this current state comes back fully clean, I should move on to other things.

File: src/EnergyPlus/Plant/DataPlant.cc [hunk 1]
```diff
@@ -49,24 +49,14 @@
 #include <ObjexxFCL/Array.functions.hh>
 
 // EnergyPlus Headers
-#include <EnergyPlus/BranchInputManager.hh>
```
Comment (mitchute) [inline src/EnergyPlus/Plant/DataPlant.cc:52]: 👍

File: src/EnergyPlus/Plant/DataPlant.cc [hunk 2]
```diff
@@ -456,65 +441,11 @@ namespace DataPlant {
     int const TypeOf_HeatPumpEIRCooling(95);
     int const TypeOf_HeatPumpEIRHeating(96);
 
-    // Parameters for General Equipment Types
```
Comment (mitchute) [inline src/EnergyPlus/Plant/DataPlant.cc:459]: Very nice.

File: src/EnergyPlus/Plant/DataPlant.hh [hunk 3]
```diff
@@ -250,42 +244,11 @@ namespace DataPlant {
     extern int const TypeOf_HeatPumpEIRCooling;
     extern int const TypeOf_HeatPumpEIRHeating;
 
-    // Parameters for General Equipment Types
```
Comment (mitchute) [inline src/EnergyPlus/Plant/DataPlant.hh:253]: Good stuff right here.

File: src/EnergyPlus/Plant/Loop.hh [hunk 4]
```diff
@@ -109,7 +105,17 @@ namespace DataPlant {
         Real64 PressureDrop;
         bool UsePressureForPumpCalcs;
         Real64 PressureEffectiveK;
-        PlantLoopSolver::PlantLoopSolverClass loopSolver;
+        // report variables
```
Comment (mitchute) [inline src/EnergyPlus/Plant/Loop.hh:108]: Not to be pedantic, but do we need to run clang-format on any of this?

File: src/EnergyPlus/Plant/LoopSideReportVars.hh [hunk 5]
```diff
@@ -53,14 +53,7 @@ namespace DataPlant {
 
     struct LoopSideReportVars
     {
```
Comment (mitchute) [inline src/EnergyPlus/Plant/LoopSideReportVars.hh]: Can't we just delete this file?

File: src/EnergyPlus/Plant/PlantManager.hh [hunk 6]
```diff
@@ -77,34 +77,9 @@ namespace PlantManager {
     extern Array1D_int SupplySideOutletNode; // Node number for the supply side outlet
     extern Array1D_int DemandSideInletNode;  // Inlet node on the demand side
 
-    struct TempLoopData
```
Comment (mitchute) [inline src/EnergyPlus/Plant/PlantManager.hh:80]: 👍

File: src/EnergyPlus/Plant/ReportVars.hh [hunk 7]
```diff
@@ -57,25 +57,7 @@ namespace DataPlant {
     {
         // Members
         // Whole loop descriptions
-        Real64 CoolingDemand;       // Plant Loop Cooling Demand, W
```
Comment (mitchute) [inline src/EnergyPlus/Plant/ReportVars.hh:60]: Another one to delete?

File: tst/EnergyPlus/unit/PlantPipingSystemsManager.unit.cc [hunk 8]
```diff
@@ -50,11 +50,11 @@
 // Google Test Headers
 #include <gtest/gtest.h>
 
-#include "EnergyPlus/DataPlant.hh"
```
Comment (mitchute) [inline tst/EnergyPlus/unit/PlantPipingSystemsManager.unit.cc:53]: Good catch.

**Comment by mitchute** [comment]
@Myoldmopar I had a few minor questions, but overall this looks great.

**Comment by mitchute** [comment]
The branch also builds locally and runs unit tests successfully.

**Comment by Myoldmopar** [comment]
Thanks @mitchute, good catches.  I can go through and delete those files and run Clang Format on the remaining plant files.  I'll push that up later.

**Comment by Myoldmopar** [comment]
@mitchute I decided to hold off on running clang-format just yet.  I'd like to get the PR list down pretty far then re-run it on a lot of files all at once.  I'm already likely to cause at least a couple conflicts with this refactor, but at least as of right now, the conflicts should be straightforward to solve.  If clang-format gets run, it could cause more complication.  

I did clean out the two empty files, great catch on that.  I pulled develop and built locally and everything is good to go afaik.  CI can verify but other than a problem there this should be good to go in.

**Comment by mitchute** [comment]
Everything looks good. Merging.

**Pull request merged**